docs/GSoC.md

```diff
@@ -0,0 +1,52 @@
+# GSoC at Mesa — Candidates Guide
+
+## What is Mesa?
+Mesa is an open-source Python library for agent-based modeling, ideal for simulating complex systems and exploring emergent behaviors. It lets you quickly build models using core components (e.g., spatial grids, schedulers) or custom implementations, visualize them in a browser, and analyze results with Python’s data tools. The goal: make simulations accessible so we can better understand and tackle complex problems.
+
+## Mesa and GSoC
+Mesa has participated in Google Summer of Code since 2024 and has had 7 contributors to date.
+
+**Last year’s projects (4) :**
+
+| Project | Contributor | Discussion Link |
+|----------|--------------|-----------------|
+| **Mesa Frames upgrade** | [Ben](https://github.com/Ben-geo) | [Discussion #151](https://github.com/projectmesa/mesa-frames/discussions/151) |
+| **Mesa front-end upgrade** | [Sahil](https://github.com/Sahil-Chhoker) | [Discussion #2772](https://github.com/projectmesa/mesa/discussions/2772) |
+| **Mesa LLM** | [Colin](https://github.com/colinfrisch) | [Discussion #2773](https://github.com/projectmesa/mesa/discussions/2773) |
+| **Mesa LLM** | [Sanika](https://github.com/sanika-n) | [Discussion #2775](https://github.com/projectmesa/mesa/discussions/2775) |
+
+Note : for the Mesa LLM project, the two candidates were selected to work on different aspects of the project
+
+
+## Making a proposal for Mesa
+
+You can start with the official GSoC guidelines on the [proposals](https://google.github.io/gsocguides/student/writing-a-proposal).
+
+For Mesa, making a great proposal implies:
+
+- Referring in a very specific way to current Mesa features and demonstrating hands-on experience with Mesa. Don’t hesitate to visit the [Mesa GitHub](https://github.com/projectmesa/mesa) and [documentation](https://mesa.readthedocs.io/latest/).
```

Comment:
```suggestion
- Building on current Mesa features and demonstrating how it will integrate with them. Hands-on experience with Mesa and building ABMs helps a lot with that. Don’t hesitate to visit the [Mesa GitHub](https://github.com/projectmesa/mesa) and [documentation](https://mesa.readthedocs.io/latest/).
```